cabot/cabotapp/models.py

```diff
@@ -349,16 +350,20 @@ def is_silenced(self, now=None):
 
 class Service(CheckGroupMixin):
 
+    @transaction.atomic()
     def update_status(self):
+        vals = Service.objects.select_for_update().filter(pk=self.pk).values_list('deleted', flat=True)
+        if len(vals) == 0 or vals[0]:
```

Comment:
in what situations would len(vals) be 0?

Reply:
If the service was deleted between this function being called and the "get deleted" call (unlikely, but I think it's possible). I think this was added out of paranoia, I'm open to removing it.

Reply:
No longer need to do this.